src/Avatar.tsx

```diff
@@ -99,10 +99,10 @@ export const Avatar: React.FC<Props> = ({
     [size]
   );
 
-  const resolvedSrc = useMemo(
-    () => resolveAvatarSrc(client, src, sizePx),
-    [client, src, sizePx]
-  );
+  const resolvedSrc = useMemo(() => {
+    if (!client || !src || !sizePx) return undefined;
+    return resolveAvatarSrc(client, src, sizePx);
```

Comment:
Ah, but the type on the `resovleAvatarSrc` was not optional (though it used optional checks inside the function 🙂), so the type error was triggered and I did not check how the function looks. Now, after looking, I realized that the function could be inlined since everything is checked. Thanks for pointing this out!